src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldSymbol.cs

```diff
@@ -314,160 +115,13 @@ internal sealed override void DecodeWellKnownAttribute(ref DecodeWellKnownAttrib
-                    }
-
-                    if (data.ConstValue == CodeAnalysis.ConstantValue.Unset)
-                    {
-                        data.ConstValue = attrValue;
-                    }
-                }
-                else
-                {
-                    constValue = data.ConstValue;
-
-                    if (constValue != CodeAnalysis.ConstantValue.Unset)
-                    {
-                        if (constValue != attrValue)
-                        {
-                            arguments.Diagnostics.Add(ErrorCode.ERR_FieldHasMultipleDistinctConstantValues, arguments.AttributeSyntaxOpt.Location);
-                        }
-                    }
-                    else
-                    {
-                        data.ConstValue = attrValue;
-                    }
-                }
-            }
-        }
-
-        internal override void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttributeData> boundAttributes, ImmutableArray<AttributeSyntax> allAttributeSyntaxNodes, DiagnosticBag diagnostics, AttributeLocation symbolPart, WellKnownAttributeData decodedData)
-        {
-            Debug.Assert(!boundAttributes.IsDefault);
-            Debug.Assert(!allAttributeSyntaxNodes.IsDefault);
-            Debug.Assert(boundAttributes.Length == allAttributeSyntaxNodes.Length);
-            Debug.Assert(_lazyCustomAttributesBag != null);
-            Debug.Assert(_lazyCustomAttributesBag.IsDecodedWellKnownAttributeDataComputed);
-            Debug.Assert(symbolPart == AttributeLocation.None);
 
-            var data = (CommonFieldWellKnownAttributeData)decodedData;
-            int? fieldOffset = data != null ? data.Offset : null;
-
-            if (fieldOffset.HasValue)
-            {
-                if (this.ContainingType.Layout.Kind != LayoutKind.Explicit)
-                {
-                    Debug.Assert(boundAttributes.Any());
-
-                    // error CS0636: The FieldOffset attribute can only be placed on members of types marked with the StructLayout(LayoutKind.Explicit)
-                    int i = boundAttributes.IndexOfAttribute(this, AttributeDescription.FieldOffsetAttribute);
-                    diagnostics.Add(ErrorCode.ERR_StructOffsetOnBadStruct, allAttributeSyntaxNodes[i].Name.Location);
-                }
-            }
-            else if (!this.IsStatic && !this.IsConst)
-            {
-                if (this.ContainingType.Layout.Kind == LayoutKind.Explicit)
-                {
-                    // error CS0625: '<field>': instance field types marked with StructLayout(LayoutKind.Explicit) must have a FieldOffset attribute
-                    diagnostics.Add(ErrorCode.ERR_MissingStructOffset, this.ErrorLocation, this);
-                }
-            }
-
-            base.PostDecodeWellKnownAttributes(boundAttributes, allAttributeSyntaxNodes, diagnostics, symbolPart, decodedData);
+            base.DecodeWellKnownAttribute(ref arguments);
```

Comment:
>base.DecodeWellKnownAttribute(ref arguments); [](start = 12, length = 45)

Should we put this into ```else```?

Reply:
Sure, that makes sense. Thanks

---
In reply to: [149510032](https://github.com/dotnet/roslyn/pull/22664#discussion_r149510032) [](ancestors = 149510032)